src/wp-includes/comment.php

```diff
@@ -1616,12 +1617,38 @@ function wp_trash_comment( $comment_id ) {
 		 */
 		do_action( 'trashed_comment', $comment->comment_ID, $comment );
 
+		// For top level 'note' type comments, also trash any children as well.
+		if ( 'note' === $comment->comment_type && 0 === (int) $comment->comment_parent ) {
+			wp_trash_comment_children( $comment->comment_ID );
+		}
+
 		return true;
```

Comment:
What happens if some children are trashed but not all? I think that as is, the status would always be `true`. I think that's fine because the top-level comment was trashed, which is the ID passed to the function.

I don't think we need to address this now, but it will likely come up in the future.